### PR TITLE
runtime-rs: Improve Cloud Hypervisor config handling

### PR DESCRIPTION
**Pull request by jodh-intel**
Replace `cloud_hypervisor_vm_create_cfg()` with a set of `TryFrom` trait implementations in the new CH specific `convert.rs` to allow the generic `Hypervisor` configuration to be converted into the CH specific `VmConfig` type.

Note that device configuration is not currently handled in `convert.rs` (it's handled in `inner_device.rs`).

This change removes the old hard-coded CH specific configuration.

Fixes: #6203.

### CODE REVIEW

**Comment by jodh-intel** [comment]
metrics CI passed, but failed to update the GH status:

```
19:38:00 Could not update commit status of the Pull Request on GitHub.
19:56:54 org.kohsuke.github.HttpException: {"message":"API rate limit exceeded for user ID 
```

@GabyCT - Have you seen this issue recently?

The arm + ppc CI's seem unwell (arm job doesn't exist and ppc shows a spectacular set of java exceptions which I think can be summarised as: PR directory `ENOENT`). Let's try again...

/retest-ubuntu-metrics
/retest-arm
/retest-power

**Comment by jodh-intel** [comment]
/retest-power

@jongwu - is the ARM CI down currently?

**Comment by jongwu** [comment]
Hi @jodh-intel -, Arm CI works. see the unit test error:

> 08:56:50 error[E0560]: struct `CpuFeatures` has no field named `amx`
08:56:50    --> crates/hypervisor/ch-config/src/convert.rs:143:42
08:56:50     |
08:56:50 143 |         let cpu_features = CpuFeatures { amx };
08:56:50     |                                          ^^^ `CpuFeatures` does not have this field
08:56:50 
08:56:50    Compiling persist v0.1.0 (/home/jenkins/workspace/kata-containers-2.0-ubuntu20.04-ARM-unit/go/src/github.com/kata-containers/kata-containers/src/runtime-rs/crates/persist)

**Comment by jodh-intel** [comment]
@jongwu - Thanks! I was seeing an http 404 message for some reason.

Branch updated.

/test

**Review by GabyCT — APPROVED** [review]

File: src/runtime-rs/crates/hypervisor/ch-config/src/lib.rs [hunk 1]
```diff
@@ -479,3 +481,15 @@ fn usize_is_zero(v: &usize) -> bool {
...
+
+// Type used to simplify conversion from a generic Hypervisor config
+// to a CH specific VmConfig.
+#[derive(Debug, Clone)]
+pub struct NamedHypervisorConfig {
+    pub kernel_params: String,
+    pub sandbox_path: String,
+    pub vsock_socket_path: String,
+    pub cfg: HypervisorConfig,
+
```
Comment (dborquez) [inline src/runtime-rs/crates/hypervisor/ch-config/src/lib.rs:493]: thanks @jodh-intel , if there's not reason to don't save space here, could you remove this blank line?

**Review by dborquez — CHANGES_REQUESTED** [review]

**Review by dborquez — APPROVED** [review]

File: src/runtime-rs/crates/hypervisor/ch-config/src/lib.rs (hunk 1 above)
Comment (jodh-intel) [reply src/runtime-rs/crates/hypervisor/ch-config/src/lib.rs:493]: Good catch @dborquez! @amshinde or I will remove that in the next PR :smile:

**Pull request merged**